### ISSUE TITLE
Memory layout of generated header structs is incorrect

### ISSUE THREAD

**Issue by babbaj** [Perfare/Il2CppDumper#315]
- Il2CppDumper version
6.2.1 (same behavior as 6.3.1)

- Target Unity version
2019.3.5f1

- Describe the issue
The memory layout of generated header structs does not always correctly match the layout of classes where inheritance is involved.
It looks like the problem is that the layout for non standard layout types. https://gcc.godbolt.org/z/wckCoz (this is specifically what was causing problems for me)
The offset comments in the dump.cs are correct. 
Adding padding bytes should be enough to fix this.

- Upload executable file and global-metadata.dat
gameassembly: https://l.trs.tn/SckzpPMESWYX.dll
metadata: https://l.trs.tn/mFAYMeD57jEm.dat

**Comment by Perfare** [comment]
I adjusted the output method of the field, now the fields of the parent and child classes will be placed in separate structures, this problem should have been solved.
You can download the last release version, if everything is ok, please let me know.

**Comment by babbaj** [comment]
Definitely not ideal but i planned on making my own header generator anyways.
Manually figuring out alignment and member offsets should be pretty easy so I might make a pr that goes back to the old code but adds padding members.

**Comment by Perfare** [comment]
I think the problem depends on the compiler.
The problem should be solved by adding `__declspec(align(8))` before each *_Fields, on line 743 of ScriptGenerator.cs.
Manually calculating the offset is probably not feasible, because for the generic class il2cpp does not include the field offset.

**Comment by babbaj** [comment]
I actually just had a good idea for how to fix this.
Instead of trying to match the offset of members with the offset il2cpp gives, you can simply fill the tail padding bytes with a char[] member. This essentially does the same thing as the current solution but allows you to have the header generated as it was before.

**Comment by babbaj** [comment]
I implemented this in my fork and it seems to work
https://github.com/babbaj/Il2CppDumper/blob/729749d7d4382b8af0de8189f20a493ddd671980/Il2CppDumper/Outputs/CoolHeaderGenerator.cs#L408

**Comment by Perfare** [comment]
I checked your code, and I noticed that you inherit the "_o" structure of the subclass, which will cause the klass of all subclasses to be the same as the parent class.
I split fields and used inheritance structure, and use __declspec on the Fields structure of the parent class to solve the alignment problem. Because after my test, the alignment problem only occurs on the msvc compiler, so I think this is enough.
